Source/DataModels/Resources/ResourceNavMesh.cpp

```diff
@@ -0,0 +1,1032 @@
+		if (!tile->header)
+			continue;
+
+		tc->calcTightTileBounds(tile->header, bmin, bmax);
+
+		const unsigned int col = duIntToCol(i, 64);
+		duCalcBoxColors(fcol, col, col);
+		duDebugDrawBox(dd, bmin[0], bmin[1], bmin[2], bmax[0], bmax[1], bmax[2], fcol);
+	}
+
+	for (int i = 0; i < tc->getTileCount(); ++i)
+	{
+		const dtCompressedTile* tile = tc->getTile(i);
+		if (!tile->header)
+			continue;
+
+		tc->calcTightTileBounds(tile->header, bmin, bmax);
+
+		const unsigned int col = duIntToCol(i, 255);
+		const float pad = tc->getParams()->cs * 0.1f;
+		duDebugDrawBoxWire(
+			dd, bmin[0] - pad, bmin[1] - pad, bmin[2] - pad, bmax[0] + pad, bmax[1] + pad, bmax[2] + pad, col, 2.0f);
+	}
+}
+
+void DrawObstacles(duDebugDraw* dd, const dtTileCache* tc)
+{
+	// Draw obstacles
+	for (int i = 0; i < tc->getObstacleCount(); ++i)
+	{
+		const dtTileCacheObstacle* ob = tc->getObstacle(i);
+		if (ob->state == DT_OBSTACLE_EMPTY)
+			continue;
+		if (!ob->mustBeDrawnGizmo)
+			continue;
+		float bmin[3], bmax[3];
+		tc->getObstacleBounds(ob, bmin, bmax);
+
+		unsigned int col = 0;
+		if (ob->state == DT_OBSTACLE_PROCESSING)
+		{
+			col = duRGBA(255, 255, 0, 128);
+		}
+		else if (ob->state == DT_OBSTACLE_PROCESSED)
+		{
+			col = duRGBA(255, 192, 0, 192);
+		}
+		else if (ob->state == DT_OBSTACLE_REMOVING)
+		{
+			col = duRGBA(220, 0, 0, 128);
+		}
+
+		if (ob->type == ObstacleType::DT_OBSTACLE_CYLINDER)
+		{
+			duDebugDrawCylinder(dd, bmin[0], bmin[1], bmin[2], bmax[0], bmax[1], bmax[2], col);
+			duDebugDrawCylinderWire(dd, bmin[0], bmin[1], bmin[2], bmax[0], bmax[1], bmax[2], duDarkenCol(col), 2);
+		}
+		else
+		{
+			// AABB localAABB = {{bmin[0], bmin[1], bmin[2]}, {bmax[0], bmax[1], bmax[2]}};
+			// OBB worldOBB = OBB(localAABB);
+			////float3 eulerAngle = float3(0, ob->orientedBox.yRadian, 0);
+			// worldOBB.Transform(Quat::FromEulerXYZ(0, ob->orientedBox.yRadian, 0));
+			// AABB worldAABB = worldOBB.MinimalEnclosingAABB();
+			// bmin[0] = worldAABB.minPoint[0];
+			// bmin[1] = worldAABB.minPoint[1];
+			// bmin[2] = worldAABB.minPoint[2];
+
+			// bmax[0] = worldAABB.maxPoint[0];
+			// bmax[1] = worldAABB.maxPoint[1];
+			// bmax[2] = worldAABB.maxPoint[2];
+			// TODO: PROPERLY SHOW THE DEBUG OBB. In function getObstacleBounds there are magic numbers multiplying the
+			// ABB and not applying a rotation. That's source code from the library.
+
+			const unsigned int* colConst = &col;
+			duDebugDrawBox(dd, bmin[0], bmin[1], bmin[2], bmax[0], bmax[1], bmax[2], colConst);
+			duDebugDrawBoxWire(dd, bmin[0], bmin[1], bmin[2], bmax[0], bmax[1], bmax[2], duDarkenCol(col), 2);
+		}
+	}
+}
+
+ bool ResourceNavMesh::Build(Scene* scene)
+{
+	CleanUp();
+
+	scene->UpdateSceneMeshRenderers();
+	scene->UpdateSceneBoundingBoxes();
+
+	std::vector<float> verts = scene->GetVertices();
+	std::vector<int> tris = scene->GetTriangles();
+	std::vector<float> normals = scene->GetNormals();
+
+	unsigned ntris = tris.size() / 3;
```

Comment:
```suggestion
	unsigned ntris = static_cast<unsigned>(tris.size() / 3);
```
This raises a warning as well, try with this.

Reply:
You left this @totisama